extensions/wikia/InsightsV2/InsightsV2.setup.php

```diff
@@ -0,0 +1,110 @@
+$wgAutoloadClasses['InsightsLoopController'] = $dir . 'InsightsLoopController.class.php';
+$wgAutoloadClasses['InsightsContext'] = $dir . 'InsightsContext.class.php';
+
+/**
+ * Config
+ */
+$wgAutoloadClasses['InsightsConfig'] = $dir . 'helpers/InsightsConfig.php';
+
+/**
+ * Helpers
+ */
+$wgAutoloadClasses['InsightsHelper'] = $dir . 'InsightsHelper.php';
+$wgAutoloadClasses['InsightsPaginator'] = $dir . 'helpers/InsightsPaginator.php';
+$wgAutoloadClasses['InsightsSorting'] = $dir . 'helpers/InsightsSorting.php';
+$wgAutoloadClasses['InsightsCache'] = $dir . 'helpers/InsightsCache.php';
+$wgAutoloadClasses['InsightsPageViews'] = $dir . 'helpers/InsightsPageViews.php';
+$wgAutoloadClasses['InsightsItemData'] = $dir . 'helpers/InsightsItemData.php';
+
+/**
+ * Special pages
+ */
+$wgSpecialPages['Insights'] = 'InsightsController';
+$wgSpecialPageGroups['Insights'] = 'wikia';
+
+/**
+ * Permissions
+ */
+$wgAvailableRights[] = 'insights';
+$wgGroupPermissions['*']['insights'] = true;
+
+/**
+ * Models
+ */
+$wgAutoloadClasses['InsightsModel'] = $dir . 'models/InsightsModel.php';
+$wgAutoloadClasses['InsightsQueryPageModel'] = $dir . 'models/InsightsQueryPageModel.php';
+$wgAutoloadClasses['InsightsDeadendModel'] = $dir . 'models/InsightsDeadendModel.php';
+$wgAutoloadClasses['InsightsUncategorizedModel'] = $dir . 'models/InsightsUncategorizedModel.php';
+$wgAutoloadClasses['InsightsUnconvertedInfoboxesModel'] = $dir . 'models/InsightsUnconvertedInfoboxesModel.php';
+$wgAutoloadClasses['InsightsWantedpagesModel'] = $dir . 'models/InsightsWantedpagesModel.php';
+$wgAutoloadClasses['InsightsWithoutimagesModel'] = $dir . 'models/InsightsWithoutimagesModel.php';
+
+/**
+ * Counting service
+ */
+$wgAutoloadClasses['InsightsCountService'] = $dir . 'services/InsightsCountService.class.php';
+$wgAutoloadClasses['InsightsCountApiController'] = $dir . 'controllers/InsightsCountApiController.class.php';
+
+/**
+ * The right rail module
+ */
+$wgAutoloadClasses['InsightsModuleController'] = $IP.'/skins/oasis/modules/InsightsModuleController.class.php';
+
+/**
+ * Hooks
+ */
+$wgAutoloadClasses['InsightsHooks'] = $dir . 'InsightsHooks.class.php';
+$wgHooks['BeforePageDisplay'][] = 'InsightsHooks::onBeforePageDisplay';
+$wgHooks['ArticleUpdateBeforeRedirect'][] = 'InsightsHooks::AfterActionBeforeRedirect';
+$wgHooks['ArticleCreateBeforeRedirect'][] = 'InsightsHooks::AfterActionBeforeRedirect';
+$wgHooks['MakeGlobalVariablesScript'][] = 'InsightsHooks::onMakeGlobalVariablesScript';
+$wgHooks['GetRailModuleList'][] = 'InsightsHooks::onGetRailModuleList';
+$wgHooks['wgQueryPages'][] = 'InsightsHooks::onwgQueryPages';
+$wgHooks['AfterUpdateSpecialPages'][] = 'InsightsHooks::onAfterUpdateSpecialPages';
+
+$wgExtensionFunctions[] = 'wfInsightsSetup';
+function wfInsightsSetup() {
+	global $wgRequest, $wgHooks;
+
+	if ( !empty( $wgRequest->getVal( 'insights', null ) ) ) {
+		$wgHooks['GetLocalURL'][] = 'InsightsHooks::onGetLocalURL';
+	}
```

Comment:
Minor, but I think it might be slightly cleaner to set a init method in the InsightsHooks classes that registers the hooks and has this logic, rather than having logic and global function in the setup file itself.